recipe/patches/fdcache_entity.h.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/src/fdcache_entity.h b/src/fdcache_entity.h
+index 6416b01..d2776b3 100644
+--- a/src/fdcache_entity.h
++++ b/src/fdcache_entity.h
+@@ -18,6 +18,10 @@
+  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+  */
+ 
++// Constants are defined in glibc >= 2.18
++#define FALLOC_FL_KEEP_SIZE    0x1
++#define FALLOC_FL_PUNCH_HOLE   0x2
++
+ #ifndef S3FS_FDCACHE_ENTITY_H_
+ #define S3FS_FDCACHE_ENTITY_H_
+ 
```

Comment:
I deleted the other patches and added this patch.

Reply:
I am not an expert, but according to 
https://docs.huihoo.com/doxygen/linux/kernel/3.7/uapi_2linux_2falloc_8h.html 
it seems correct.

just wondering if we should have `ifndef` in the case that `FALLOC_FL_KEEP_SIZE` and `FALLOC_FL_PUNCH_HOLE` are already defined.

Reply:
Thanks @xmnlab for the feedback. I added `ifndef` in the following commit. Do you think we can merge now?